### PR TITLE
Add webhook signature section

### PR DESCRIPTION
**Pull request by antulik**
Related https://github.com/splitpayments/split/pull/994

127 0 0 1_4567__shell

127 0 0 1_4567__shell 1

### CODE REVIEW

File: source/index.html.md [hunk 1]
```diff
@@ -464,6 +464,48 @@ All events posted to the designated URL fit the same structure.
 <aside class="notice">The sandbox environment allow both HTTP and HTTPS webhook URLs. The live environment however will only POST to HTTPS URLs.
 </aside>
 
+### Checking Webhook Signatures
+
+Split signs the webhook events it sends to your endpoints. We do so by including a signature in each event’s `Split-Signature` header. This allows you to validate that the events were sent by Split, not by a third party.
```
Comment (trev) [inline source/index.html.md]: Consider swapping:

> This allows you to validate that the events were sent by Split, not by a third party.

to

> This allows you to validate that the events were indeed sent by Split.

File: source/index.html.md [hunk 2]
```diff
@@ -464,6 +464,48 @@ All events posted to the designated URL fit the same structure.
...
+
+Before you can verify signatures, you need to retrieve your endpoint’s secret from your Webhooks settings. Each secret is unique to the endpoint to which it corresponds. If you use multiple endpoints, you must obtain a secret for each one.
```
Comment (trev) [inline source/index.html.md]: Consider swapping:

> Each secret is unique to the endpoint to which it corresponds...

to

> Each endpoint has its own unique secret; if you use multiple endpoints, you must obtain a secret for each one.

File: source/index.html.md [hunk 3]
```diff
@@ -464,6 +464,48 @@ All events posted to the designated URL fit the same structure.
...
+| `signature` | Request signature |
+| `other`     | Placeholder for the future parameters (currently not used) |
+
+**Step 2: Prepare the signed_payload string**
+
+You achieve this by concatenating:
+
+- The timestamp from the header (as a string)
+- The character `.` (dot)
+- The actual JSON payload (i.e., the request’s body)
```
Comment (trev) [inline source/index.html.md]: > (i.e., the request’s body)

to

> (request body)

File: source/index.html.md [hunk 4]
```diff
@@ -464,6 +464,48 @@ All events posted to the designated URL fit the same structure.
...
+
+**Step 1. Extract the timestamp and signatures from the header**
+
+Split the header, using the `.` (dot) character as the separator, to get a list of elements.
+
+| Element | Description |
+|---------|-------------|
+| `timestamp` | [Unix time](https://en.wikipedia.org/wiki/Unix_time) in seconds when the signature was created |
+| `signature` | Request signature |
+| `other`     | Placeholder for the future parameters (currently not used) |
```
Comment (trev) [inline source/index.html.md]: No need for the `the`

File: source/index.html.md [hunk 5]
```diff
@@ -464,6 +464,48 @@ All events posted to the designated URL fit the same structure.
...
+
+You achieve this by concatenating:
+
+- The timestamp from the header (as a string)
+- The character `.` (dot)
+- The actual JSON payload (i.e., the request’s body)
+
+**Step 3: Determine the expected signature**
+
+Compute an HMAC with the SHA256 hash function. Use the endpoint’s signing secret as the key, and use the `signed_payload` string as the message.
```
Comment (trev) [inline source/index.html.md]: Might be worth referencing https://www.jokecamp.com/blog/examples-of-creating-base64-hashes-using-hmac-sha256-in-different-languages to help get people started.

**Review by trev — APPROVED** [review]
Just a few little nitpicks. Otherwise looks 👌🏽

**Comment by antulik** [comment]
Added examples for the each language

image

image

image

image

image

**Pull request merged**